mangoPURE/data/transforms.py

```diff
@@ -126,30 +130,51 @@ def __call__(
     ) -> MixedExample:
         """
         The same as AddRandomSpeakerSegment, but it generates
-        several (random number) speakers in the audio
+        several (random number) speakers to the audio
         """
         speakers_num = random.randint(self.min_speakers, self.max_speakers)
         for i in range(speakers_num):
             example = super().__call__(example)
         return example
 
 
+class MergeAll(MixerTransform):
+    def __call__(
+            self,
+            example: MixedExample
+    ) -> MixedExample:
+        """
+        Merges all data created by previous transforms into single audio
+        """
+        if example._base is None:
+            raise Exception("the base audio was not created")
+        example.audio = example._base.audio
+        has_speaker = example._base.type == "speaker"
+        if example._segments is not None:
+            for segment in example._segments:
+                if segment.type == "speaker":
+                    example.audio[segment.start:segment.start + segment.audio.shape[0]] += segment.audio
+                    has_speaker = True
+            for segment in example._segments:
+                if segment.type == "noise":
+                    noise_coef = calc_audio_adjustment_coef(example.audio, segment.audio, segment.metadata["snr"])\
```

Comment:
We add noises one after another and this breaks real snr. I am not sure if it matters, but maybe we first merge all speakers in normal track, all noises to another track, calculate sir once and merge them. Or calculate sir for each noise relative to speakers-only audio

Reply:
Okay, I ' ll handle